lib/cosmos/io/json_drb.rb

```diff
@@ -194,13 +195,13 @@ def self.receive_message(socket, data)
     def self.get_at_least_x_bytes_of_data(socket, current_data, required_num_bytes)
       while (current_data.length < required_num_bytes)
         begin
-          data = socket.recv_nonblock(65535)
-          if data.length == 0
+          socket.read_nonblock(65535, @@data)
```

Comment:
Also need to know if data is empty after the recv_nonblock.  A class variable might already have data in it.